src/compiler/transformers/ts.ts

```diff
@@ -419,6 +421,26 @@ namespace ts {
         }
 
         /**
+         * Tests whether we should emit a __decorate call for a class declaration.
+         */
+        function shouldEmitDecorateCallForClass(node: ClassDeclaration) {
+            if (node.decorators && node.decorators.length > 0) {
+                return true;
+            }
+
+            const constructor = getFirstConstructorWithBody(node);
+            if (constructor) {

```

Comment:
change to `if (constructor && foreach(constructor.parameters, p => p.decorators && p.decorators.length > 0))`